### ISSUE TITLE
Possible Treaties made by enemy AIs

### ISSUE THREAD

**Issue by KingCaliph** [Sh0wdown/UnofficialCrusaderPatch#526]
I'll keep this short and sweet. I have thought about this since the beginning of the game, and I can finally suggest it now that I found this wonderful forum. 

I feel that a cool feature to be added would be one where enemies, whether human/AI or AI/AI, can make treaties with each other. The treaties would almost always be made if one enemy is about to die. The treaty could either be a plea, e.g. Snake is about to die so he sends a treaty, or be a form of payment, e.g. instead of Abbot killing Rat he forms a treaty with him to work together and kill Caliph. There could also be "Shows of good faith", e.g. a human player sends 2000 gold pieces with a treaty, the receiving AI could take the gold and reject the treaty or accept both. This would add a new spin to the game and make it more like historical battles. 

There could also be a feature where an ally could leave one side and join another, e.g. the Caliph leaves your side because you haven't help him in any way and he joins Snake.

**Comment by Lolasik011** [comment]
> I'll keep this short and sweet. I have thought about this since the beginning of the game, and I can finally suggest it now that I found this wonderful forum.
> 
> I feel that a cool feature to be added would be one where enemies, whether human/AI or AI/AI, can make treaties with each other. The treaties would almost always be made if one enemy is about to die. The treaty could either be a plea, e.g. Snake is about to die so he sends a treaty, or be a form of payment, e.g. instead of Abbot killing Rat he forms a treaty with him to work together and kill Caliph. There could also be "Shows of good faith", e.g. a human player sends 2000 gold pieces with a treaty, the receiving AI could take the gold and reject the treaty or accept both. This would add a new spin to the game and make it more like historical battles.
> 
> There could also be a feature where an ally could leave one side and join another, e.g. the Caliph leaves your side because you haven't help him in any way and he joins Snake.


I'm not sure that's possible... 
But the idea is definitely Amazing !

**Comment by Starshine575** [comment]
Yeah it is kinda complex but espacally the last part is for sure do able if someone is taking His/her time and has the Knowledge 👍

**Comment by KingCaliph** [comment]
@Lolasik011 You are probably right, it is most likely impossible. I mainly just wanted to share the idea and see what people thought of it.

**Comment by LordHansCapon** [comment]
It's far from impossible, not sure what you are all talking about. There are certain points where the game  can be hooked to achieve such behaviour. It is possible to set party groups while playing, it is also possible to hook into certain functions in the game where we would check certain conditions for a treaty between AIs. The hard part comes when the player recieves a treaty, or rather, when the player wants to send a treaty. The UI should work like the Ally goods send/recieve panels, and this can be solved too.

So the only thing remaining is the Time. Who would spend days on creating such mod? As a side note, AIs have no such messages, so it'd be text based, what a bummer.

**Comment by ByBurton** [comment]
It should also be adjustable for each lord what conditions there are for leaving a treaty and / or making a new one. And what treaties they accept.
E. G. Wolf would never join the player, caliph would never join an infidel and leave any treaty if he has a crusader ally.

**Comment by KingCaliph** [comment]
@ByBurton EXACTLY. I forgot to add that part. I actually made a long list of conditions for each AI but just forgot to add it. However, the more conditions the harder it becomes, especially since some of the of them might be impossible.

**Comment by Starshine575** [comment]
Although it is a really cool Idea it is nothing that I'd like to see in the patch in near furture. There are some other problems (espacally with the AI) that should be have a higher priority IMHO. 😅😁

**Comment by KingCaliph** [comment]
@Starshine575 true. there are so many cools features/ bug fixes that need to be addressed first. the treaty idea was just something I thought of and wanted to see what people thought.

**Comment by Starshine575** [comment]
Yeah it is also a very cool Idea, nice you mentioned it. 😊

**Comment by Krarilotus** [comment]
#197 just wanted to mention this has been thought of somewhat before.

**Comment by KingCaliph** [comment]
@Krarilotus Oh, opps. I looked around for awhile and I didn't see anything about the subject but clearly I missed this one, probably because the first part was in German so I didn't bother to look at the rest.
